src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -118,31 +118,34 @@ public async Task<IEnumerable<ConcurrencyManagerResult<SimulationResults>>> RunC
             throw new OSPSuiteException(Error.InvalidMixOfSimulationAndSimulationBatch);
 
          _cancellationTokenSource = new CancellationTokenSource();
+
          if (_simulations.Count > 0)
          {
-            var results = await _concurrencyManager.RunAsync(
-               numberOfCores(),
+            var results = new ConcurrentDictionary<IModelCoreSimulation, ConcurrencyManagerResult<SimulationResults>>();
+            await _concurrencyManager.RunAsync(
```

Comment:
let's have the concurrentManager create the results so that you can always return